### PR TITLE
Add case insensitive support to Parquet.

### PR DESCRIPTION
**Pull request by rdblue**
#83 and #89 add case sensitivity flags, but filters pushed down to Parquet still do not support case sensitivity. This updates the Parquet row group filters to support a case sensitivity flag and updates Spark to pass the flag.

This also updates the old read path that uses `ReadSupport` when converting Iceberg expressions to Parquet filters.

### CODE REVIEW

**Comment by rdblue** [comment]
@xabriel, can you review? We ran into this when testing case insensitive queries.

File: spark/src/test/java/com/netflix/iceberg/spark/data/TestParquetAvroReader.java [hunk 1]
```diff
@@ -101,10 +99,11 @@ public void testStructSchema() throws IOException {
...
 
-      try (ParquetReader<Record> reader = new ParquetReader<>(
-          Files.localInput(testFile), structSchema, ParquetReadOptions.builder().build(),
-          fileSchema -> ParquetAvroValueReaders.buildReader(structSchema, readSchema),
-          Expressions.alwaysTrue(), true)) {
+      try (CloseableIterable<Record> reader = Parquet.read(Files.localInput(testFile))
+          .project(structSchema)
+          .createReaderFunc(
+              fileSchema -> ParquetAvroValueReaders.buildReader(structSchema, readSchema))
+           .build()) {
```
Comment (xabriel) [inline spark/src/test/java/com/netflix/iceberg/spark/data/TestParquetAvroReader.java]: These changes to `TestParquetAvroReader.java` seem unrelated to PR intention. Can you explain?

Comment (rdblue) [reply spark/src/test/java/com/netflix/iceberg/spark/data/TestParquetAvroReader.java]: Rather than updating the ParquetReader call to include the new boolean for case sensitivity, I moved them to use the builder instead. It fixes the problem this time around, and makes sure it won't fail other tests in the future.

File: parquet/src/main/java/com/netflix/iceberg/parquet/ParquetMetricsRowGroupFilter.java [hunk 2]
```diff
@@ -56,9 +56,13 @@ private MetricsEvalVisitor visitor() {
...
 
   public ParquetMetricsRowGroupFilter(Schema schema, Expression unbound) {
+    this(schema, unbound, true);
+  }
+
+  public ParquetMetricsRowGroupFilter(Schema schema, Expression unbound, boolean caseSensitive) {
     this.schema = schema;
     this.struct = schema.asStruct();
-    this.expr = Binder.bind(struct, rewriteNot(unbound), true);
+    this.expr = Binder.bind(struct, rewriteNot(unbound), false);
```
[the remark this answers is not in the document]
Comment (rdblue) [reply parquet/src/main/java/com/netflix/iceberg/parquet/ParquetMetricsRowGroupFilter.java]: Yep. Good catch!

**Comment by rdblue** [comment]
@xabriel, thanks for looking. You're right that I still need to add tests.

File: parquet/src/main/java/com/netflix/iceberg/parquet/ParquetMetricsRowGroupFilter.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (rdblue) [reply parquet/src/main/java/com/netflix/iceberg/parquet/ParquetMetricsRowGroupFilter.java]: I've updated this and fixed the test. The problem was that it used a `notEq` predicate that always matches. Now it uses an `eq` predicate that has a different result than for a column that isn't found.

**Comment by rdblue** [comment]
@xabriel, can you take another look? I've fixed the problem you found and validated there are tests that actually exercise case insensitive binding for the Parquet classes.

To your question about how to reproduce this, you have to run a query that pushes filters down to Parquet. The filters you use can't be satisfied by partition filtering because Iceberg will drop those filters and won't push them down. I would recommend writing a dataset without partitioning and sorted by a categorical value, then test equality filters on that categorical value. That would hit this problem.

**Comment by rdblue** [comment]
Thanks for reviewing this, @xabriel! I'll merge it.
